feat: snowflake streaming query

rephus opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

Partially closes: #5350

### Description:

https://docs.snowflake.com/en/developer-guide/node-js/nodejs-driver-consume#batch-processing-results

![Screenshot from 2023-05-05 12-05-22](https://user-images.githubusercontent.com/1983672/236430655-49e6b660-d455-4bfa-94df-276b7fea7415.png)

<!-- Add a description of the changes proposed in the pull request. -->

<!-- Even better add a screenshot / gif / loom -->

ZeRego commented:
@rephus In your screenshot the date doesn't seem to match our formatting 🤔 

![Captura de ecrã 2023-05-05, às 13 51 36](https://user-images.githubusercontent.com/9117144/236462988-24858d0d-7d32-4cba-bdaa-e9b9ac36d049.png)

Can you double check the `parceCell` is working?

rephus commented:
> @rephus In your screenshot the date doesn't seem to match our formatting 🤔

Good catch. This was because the types were not added correctly when I added the project. 

I fixed it by doing `lightdash generate` on the CLI and now it looks good 

<img width="1191" alt="image" src="https://user-images.githubusercontent.com/1983672/236471960-251b347e-62b5-4838-ba86-f8a6ec42b9e0.png">

rephus commented:
btw I made a few changes to the code, I only apply the streaming when fetching results, we use the same method in multiple places

rephus commented on packages/warehouses/src/warehouseClients/SnowflakeWarehouseClient.ts:
```diff
@@ -183,6 +182,48 @@ export class SnowflakeWarehouseClient extends WarehouseBaseClient<CreateSnowflak
+                        }),
+                        {},
+                    );
+                    stmt.streamRows()
```
because this is how it is done on snowflake docs https://docs.snowflake.com/en/developer-guide/node-js/nodejs-driver-consume#batch-processing-results

rephus replied:
Done
